OpenTelemetry.proj

```diff
@@ -3,9 +3,6 @@
     <SolutionProjects Include="**\*.csproj" />
     <PackProjects Include="src\**\*.csproj" />
     <TestProjects Include="test\**\*.csproj" />
-
-    <!-- Not pack SemanticConventions project for now -->
-    <PackProjects Remove="src\OpenTelemetry.SemanticConventions\OpenTelemetry.SemanticConventions.csproj" />
```

Comment:
I'm not sure that we want to start pushing new releases of this package. The conventions that it generates appear to be very out of date. There's new tooling that is probably worth studying before we start publishing this package (see #5156).

I think we should unlist the version that was published a long while back. I think it was a mistake. https://www.nuget.org/packages/OpenTelemetry.SemanticConventions

Reply:
+1

Reply:
Nothing on here is going to cause this package to be published. This line is just all kinds of wrong. If we wanted to keep this project from being packaged it is `<IsPackable>false</IsPackable>`. But there's no harm in it being packaged. We package lots of stuff and only manually pick what to publish. In the release doc I say we'll only push this `OpenTelemetry.SemanticConventions` if we decide to do so. It isn't currently part of the release cadence defined for everything else.

Now what to do with it...

I've been messing with it in a branch trying to generate something which will be actually useful. Similar to #5156 but with some changes to the generator code as well.

I would like to land this PR defining the process. Then we can have follow-up discussions about what we want to do with `OpenTelemetry.SemanticConventions`. Perhaps land an improved version or those discussions may end up we just delete it. What I'm trying to do here is move us from undefined state to defined state for everything currently in the repo and then we can refine from there.

Reply:
My suggest - for this PR, either leave `OpenTelemetry.SemanticConventions` untouched, or remove it from the release process.

Reply:
This is what I think this change does...

By removing these two lines, the OpenTelemetry.SemanticConventions project will now get packed when our PublishPackages job runs

https://github.com/open-telemetry/opentelemetry-dotnet/blob/57c442cc7b2fe93c81ca877de28f5204fdb11ebb/.github/workflows/publish-packages-1.0.yml#L32-L45

The concern is that because the push to NuGet is a manual step, it is easy for us to accidentally push the OpenTelemetry.SemanticConventions package. I suspect this is how it was accidentally released in the past.

Correct me if I'm wrong about any of this. My main concern is that I just want to avoid human error.

Reply:
If you create a release today for _any_ tag you get these packages:

`OpenTelemetry.Api`
`OpenTelemetry.Api.ProviderBuilderExtensions`
`OpenTelemetry`
`OpenTelemetry.Exporter.Console`
`OpenTelemetry.Exporter.InMemory`
`OpenTelemetry.Exporter.OpenTelemetryProtocol`
`OpenTelemetry.Exporter.Prometheus.AspNetCore`
`OpenTelemetry.Exporter.Prometheus.HttpListener`
`OpenTelemetry.Exporter.Zipkin`
`OpenTelemetry.Extensions.Hosting`
`OpenTelemetry.Extensions.Propagators`
`OpenTelemetry.Instrumentation.AspNetCore`
`OpenTelemetry.Instrumentation.Http`
`OpenTelemetry.Instrumentation.SqlClient`
`OpenTelemetry.Shims.OpenTracing`

You have to manually choose the package(s) to push. The whole process is rife with manual intervention and subject to human error. So my question is, what is special about `OpenTelemetry.SemanticConventions`?

I just pushed this commit: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5531/commits/2987841a1e8d65e8539d1962960ae2fbf3df3df2

What that does is align the package output with the tag which spawned the package workflow.

`core-`
`OpenTelemetry.Api`
`OpenTelemetry.Api.ProviderBuilderExtensions`
`OpenTelemetry`
`OpenTelemetry.Exporter.Console`
`OpenTelemetry.Exporter.InMemory`
`OpenTelemetry.Exporter.OpenTelemetryProtocol`
`OpenTelemetry.Exporter.Zipkin`
`OpenTelemetry.Extensions.Hosting`
`OpenTelemetry.Extensions.Propagators`

`coreunstable-`
`OpenTelemetry.Exporter.Prometheus.AspNetCore`
`OpenTelemetry.Exporter.Prometheus.HttpListener`
`OpenTelemetry.Shims.OpenTracing`
`OpenTelemetry.SemanticConventions`

`Instrumentation.AspNetCore-`
`OpenTelemetry.Instrumentation.AspNetCore`

`Instrumentation.Http-`
`OpenTelemetry.Instrumentation.Http`

`Instrumentation.SqlClient-`
`OpenTelemetry.Instrumentation.SqlClient`

This reduces the burden of intervention. This is the thinking we need to adopt. We need to move this process to one which is automated and does not require intervention and is not subject to human error 🤣

So coming back to the issue of `OpenTelemetry.SemanticConventions`. If we don't want to release or maintain `OpenTelemetry.SemanticConventions` why is it in the repo? Let's just delete it!

Reply:
> So coming back to the issue of `OpenTelemetry.SemanticConventions`. If we don't want to release or maintain `OpenTelemetry.SemanticConventions` why is it in the repo? Let's just delete it!

➕